src/benchmarks/micro/ValuesGenerator.cs

```diff
@@ -87,12 +94,34 @@ private static T GenerateValue<T>(Random random)
                 return (T)(object)random.Next();
             if (typeof(T) == typeof(double))
                 return (T)(object)random.NextDouble();
+            if (typeof(T) == typeof(float))
+                return (T)(object)(float)random.NextDouble();
             if (typeof(T) == typeof(string))
                 return (T) (object) GenerateRandomString(random, 1, 50);
             
             throw new NotImplementedException($"{typeof(T).Name} is not implemented");
         }
 
+        private static T GenerateValue<T>(Random random, T minValue, T maxValue)
```

Comment:
Many of the `System.Math` functions have special handling in the JIT to either cache computed results or to perform constant folding. So we want to generate a unique input per iteration that is within the allowed domain.

Reply:
@tannergooding Could you please add this as a function header/comment?

Reply:
Fixed.